components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -298,6 +299,9 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
         // the other scope validators.
         removeInternalScopes(tokReqMsgCtx);
 
+        // Adding the authorized internal scopes to tokReqMsgCtx as a property for any special validators to use.
+        tokReqMsgCtx.addProperty(AUTHORISED_INTERNAL_SCOPES_PARAM_NAME, authorizedInternalScopes);
```

Comment:
Instead of a property how about making this a first class attribute of the context?

Reply:
I guess will have to change the model and everything. Haven't we introduced this property field for this sort of modifications?

Reply:
TokenContext is a model used internally I guess (Not exposed directly via an admin service). For readability, I prefer to have it as a separate attribute in the context.

Otherwise we have to document this somewhere. Otherwise extension developers won't realize the internal scopes are available unless they debug and check the context properties.